embassy-nrf/src/radio/ble.rs

```diff
@@ -0,0 +1,431 @@
+        // For what I understand, this config only filter the packets
+        // by the address, so only packages send to the previous address
+        // will finish the reception
+        r.rxaddresses.write(|w| {
+            w.addr0()
+                .enabled()
+                .addr1()
+                .enabled()
+                .addr2()
+                .enabled()
+                .addr3()
+                .enabled()
+                .addr4()
+                .enabled()
+        });
+    }
+
+    /// Set the CRC polynomial
+    /// It only uses the 24 least significant bits
+    ///
+    /// The radio must be disabled before calling this function
+    pub fn set_crc_poly(&mut self, crc_poly: u32) {
+        let r = T::regs();
+
+        r.crcpoly.write(|w| unsafe {
+            // Configure the CRC polynomial
+            // Each term in the CRC polynomial is mapped to a bit in this
+            // register which index corresponds to the term's exponent.
+            // The least significant term/bit is hard-wired internally to
+            // 1, and bit number 0 of the register content is ignored by
+            // the hardware. The following example is for an 8 bit CRC
+            // polynomial: x8 + x7 + x3 + x2 + 1 = 1 1000 1101 .
+            w.crcpoly().bits(crc_poly & 0xFFFFFF)
+        });
+    }
+
+    /// Set the CRC init value
+    /// It only uses the 24 least significant bits
+    /// The CRC initial value varies depending of the PDU type
+    ///
+    /// The radio must be disabled before calling this function
+    pub fn set_crc_init(&mut self, crc_init: u32) {
+        let r = T::regs();
+
+        r.crcinit.write(|w| unsafe { w.crcinit().bits(crc_init & 0xFFFFFF) });
+    }
+
+    /// Set the radio tx power
+    ///
+    /// The radio must be disabled before calling this function
+    pub fn set_tx_power(&mut self, tx_power: TxPower) {
+        let r = T::regs();
+
+        r.txpower.write(|w| w.txpower().variant(tx_power));
+    }
+
+    /// Set buffer to read/write
+    ///
+    /// This method is unsound. You should guarantee that the buffer will live
+    /// for the life time of the transmission or if the buffer will be modified.
+    /// Also if the buffer is smaller than the packet length, the radio will
+    /// read/write memory out of the buffer bounds.
+    pub fn set_buffer(&mut self, buffer: &[u8]) -> Result<(), Error> {
+        // Because we are serializing the buffer, we should always have the buffer in RAM
+        slice_in_ram_or(buffer, Error::BufferNotInRAM)?;
+
+        let r = T::regs();
+
+        // Here we are considering that the length of the packet is
+        // correctly set in the buffer, otherwise we will sending
+        // unowned regions of memory
+        let ptr = buffer.as_ptr();
+
+        // Configure the payload
+        r.packetptr.write(|w| unsafe { w.bits(ptr as u32) });
+
+        Ok(())
+    }
+
+    /// Send packet
+    pub async fn transmit(&mut self) {
```

Comment:
Should the buffer be passed as part of transmit/receive to ensure that it lives for the duration of transmission, or is set_buffer useful outside of 'transmit/receive'?

Reply:
On legacy advertising, the radio transmits the same data multiple times across multiple channels, this ways It's possible to change set the buffer just once and make a cheaper transmission call ([example on jewel](https://github.com/jewel-rs/jewel/blob/90a2e498db4766c816cdf4f1451387756071cb05/jewel/src/ll/mod.rs#L76-L112)).
But I agree that this is a micro-optimization and unnecessary for this case and maybe we should prefer a simpler interface. 
What do you think?

Reply:
I already answered myself in my own comment, so I already made the changes.